File: python/app/unsubscribe_service.py
from app.statement_api import StatementAPI

class UnsubscribeService:
    def display_recurring_charges():
        recurring_charges = {}
        bill_statement = {}

        for charge in StatementAPI.get_charges():
            name = charge["name"]
            total = bill_statement.get(name, 0)
            total += 1
            bill_statement[name] = total

        for charge_name, charge_count in bill_statement.items():
            if charge_count >= 1:
                recurring_charges[charge_name] = charge_count

        return recurring_charges

File: java/src/main/java/UnsubscribeService.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class UnsubscribeService {
   public Map<String, Integer> displayRecurringCharges() {
        Map<String, Integer> billStatement = new HashMap<String, Integer>();
        Map<String, Integer> recurringCharges = new HashMap<String, Integer>();

        for (Map<String, String> charge : StatementApi.getCharges()) {
            String name = charge.get("name");
            int chargeCount = billStatement.getOrDefault(name, 0) + 1;
            billStatement.put(name, chargeCount);
        }

        for (Map.Entry<String, Integer> chargeEntry : billStatement.entrySet()) {
            if (chargeEntry.getValue() >= 1) {
                recurringCharges.put(chargeEntry.getKey(), chargeEntry.getValue());
            }
        }

        return recurringCharges;
    }
}
